libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -120,6 +121,12 @@ func makeReporter(beat beat.Info, settings report.Settings, cfg *common.Config)
 		return nil, err
 	}
 
+	// Unset username which is set by default, even if no password is set
+	if config.APIKey != "" {
+		config.Username = ""
+		config.Password = ""
+	}
+
```

Comment:
Yes its a bummer, we have an issue for that https://github.com/elastic/beats/issues/14226